### PR TITLE
 Core/Entities: Added ObjectGuid class from MaNGOS 

### PR DESCRIPTION
**Pull request by Viste**
##### CHANGES PROPOSED:
-  Added ObjectGuid class from MaNGOS
##### Target branch(es):master
not complited

### CODE REVIEW

**Comment by Winfidonarleyan** [comment]
Very good

**Comment by Viste** [comment]
> Very good
I'm getting my eyes blown out from some parts of the code

**Comment by FrancescoBorzi** [comment]
@Viste good job! What do you think should be checked in order to merge this? In other words, what can this potentially affect in the game (for testers)?

**Comment by Viste** [comment]
> 
> 
> @Viste good job! What do you think should be checked in order to merge this? In other words, what can this potentially affect in the game (for testers)?

all what related to guids??

File: src/server/game/Entities/Object/ObjectGuid.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+template uint32 ObjectGuidGenerator<HIGHGUID_PLAYER>::Generate();
+template uint32 ObjectGuidGenerator<HIGHGUID_GAMEOBJECT>::Generate();
+template uint32 ObjectGuidGenerator<HIGHGUID_TRANSPORT>::Generate();
+template uint32 ObjectGuidGenerator<HIGHGUID_UNIT>::Generate();
+template uint32 ObjectGuidGenerator<HIGHGUID_PET>::Generate();
+template uint32 ObjectGuidGenerator<HIGHGUID_VEHICLE>::Generate();
+template uint32 ObjectGuidGenerator<HIGHGUID_DYNAMICOBJECT>::Generate();
+template uint32 ObjectGuidGenerator<HIGHGUID_CORPSE>::Generate();
+template uint32 ObjectGuidGenerator<HIGHGUID_INSTANCE>::Generate();
+template uint32 ObjectGuidGenerator<HIGHGUID_GROUP>::Generate();
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Object/ObjectGuid.cpp]: Need to add a new line

File: src/server/game/Entities/Object/ObjectGuid.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-GPL2
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Object/ObjectGuid.cpp]: For new file need use `AGPL3`
```
Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU AGPL v3 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-AGPL3```

File: src/server/game/Entities/Object/ObjectGuid.h [hunk 3]
```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-GPL2
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Object/ObjectGuid.h]: For new file need use `AGPL3`
```
Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU AGPL v3 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-AGPL3```

File: src/server/game/Entities/Object/ObjectGuid.h [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    {
+        public:
+            size_t operator()(ObjectGuid const& key) const
+            {
+                return hash<uint64>()(key.GetRawValue());
+            }
+    };
+}
+
+#endif // ObjectGuid_h__
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Object/ObjectGuid.h]: miss new line

File: src/server/game/Entities/Unit/Unit.cpp [hunk 5]
```diff
@@ -2500,8 +2500,11 @@ void Unit::SendMeleeAttackStop(Unit* victim)
     // ClearUnitState(UNIT_STATE_MELEE_ATTACKING); // ZOMG! commented out for now
 
     WorldPacket data(SMSG_ATTACKSTOP, (8+8+4));
-    data.append(GetPackGUID());
-    data.append(victim ? victim->GetPackGUID() : 0);
+    data << GetPackGUID();
+    if (victim)
+        data << victim->GetPackGUID();
+    else
+        data << uint8(0);
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Unit/Unit.cpp]: miss new line? :D

File: src/server/game/Entities/Object/ObjectGuid.cpp (hunk 2 above)
Comment (Viste) [reply src/server/game/Entities/Object/ObjectGuid.cpp]: your license toubles not my problem

File: src/server/game/Entities/Object/ObjectGuid.h (hunk 3 above)
Comment (Viste) [reply src/server/game/Entities/Object/ObjectGuid.h]: your license toubles not my problem

File: src/server/game/Entities/Unit/Unit.cpp (hunk 5 above)
Comment (Viste) [reply src/server/game/Entities/Unit/Unit.cpp]: no

**Comment by Viste** [comment]
reopen soon need do some clean up in repo and w8 for some merges of another prs

**Comment by Helias** [comment]
We really need a PR like this, to be compatible with public TC-based patches/scripts/core.

**Comment by Viste** [comment]
where you be year ago :D

**Comment by FrancescoBorzi** [comment]
I've always said that we need this. I just don't have time to help with it too.

**Pull request closed without merging**